packages/flutter_tools/lib/src/vmservice.dart

```diff
@@ -1536,43 +1548,56 @@ class FlutterView extends ServiceObject {
     await subscription.cancel();
   }
 
-  Future<void> setAssetDirectory(Uri assetsDirectory) async {
+  bool get hasIsolate => _uiIsolate != null;
+
+  @override
+  String toString() => id;
+}
+
+/// Flutter specific VM Service functionality.
+extension FlutterVmService on vm_service.VmService {
+  /// Set the asset directory for the an attached Flutter view.
+  Future<void> setAssetDirectory({
+    @required Uri assetsDirectory,
+    @required String viewId,
+    @required String uiIsolateId,
+  }) async {
     assert(assetsDirectory != null);
-    await owner.vmService.vm.invokeRpc<ServiceObject>('_flutter.setAssetBundlePath',
-        params: <String, dynamic>{
-          'isolateId': _uiIsolate.id,
-          'viewId': id,
-          'assetDirectory': assetsDirectory.toFilePath(windows: false),
-        });
-  }
-
-  Future<void> setSemanticsEnabled(bool enabled) async {
-    assert(enabled != null);
-    await owner.vmService.vm.invokeRpc<ServiceObject>('_flutter.setSemanticsEnabled',
-        params: <String, dynamic>{
-          'isolateId': _uiIsolate.id,
-          'viewId': id,
-          'enabled': enabled,
-        });
-  }
-
-  Future<Map<String, Object>> getSkSLs() async {
-    final Map<String, dynamic> response = await owner.vmService.vm.invokeRpcRaw(
-      '_flutter.getSkSLs',
-      params: <String, dynamic>{
-        'viewId': id,
+    await callMethod(kSetAssetBundlePathmMethod,
```

Comment:
I think the cache is more trouble than it is worth, because we still need to correctly handle invalidation. This currently relies on hooking into a vm internal stream (https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/vmservice.dart#L146) to receive updates after hot restarts only, which I'd love to remove.